src/hooks/useEvents.ts

```diff
@@ -14,25 +16,33 @@ const useUpcomingEvents = () => {
     return Number(new Date(a.date)) - Number(new Date(b.date));
   };
 
+  const fetchEvents = async () => {
+    try {
+      setIsFetching(true);
+      // pass if the user is authenticated into function to get the correct return value for api
+      const eventData = await getUpcomingEvents(authenticated);
+      const sorted = [...eventData].sort(sortByDate);
+      setEvents(sorted);
+      setIsFetching(false);
+    } catch (error) {
+      setError(error.statusCode);
+    } finally {
+      setIsFetching(false);
+    }
+  };
+
   useEffect(() => {
-    const fetchEvents = async () => {
-      try {
-        setIsFetching(true);
-        // pass if the user is authenticated into function to get the correct return value for api
-        const eventData = await getUpcomingEvents(authenticated);
-        const sorted = [...eventData].sort(sortByDate);
-        setEvents(sorted);
-        setIsFetching(false);
-      } catch (error) {
-        setError(error.statusCode);
-      } finally {
-        setIsFetching(false);
-      }
-    };
     fetchEvents();
   }, [authenticated]);
 
-  return { isFetching, events, setEvents, error };
+  useEffect(() => {
```

Comment:
Can't we just return `fetchEvents` from this hook and get rid og this useEffect?.

Then we can just call `fetchEvents()` directly instead of setting a state to trur which triggers a useEffect that fetched events and immediately sets the state to false again

Reply:
The useEffect with regard to the authentication parameter is there to ensure when authentication change on the client, the correct data is collected from the API. For instance, when you are automatically logged out as admin you want to show events that should only be visible for non-auth users

Reply:
Ahhh I see, yes that could be done :+1: